main.go

```diff
@@ -277,7 +289,25 @@ func (s *TailnetSrv) listen(srv *tsnet.Server) (net.Listener, error) {
 		}
 		l, err = srv.ListenFunnel("tcp", s.ListenAddr, opts...)
 	case !s.ServePlaintext:
-		l, err = srv.ListenTLS("tcp", s.ListenAddr)
+		if s.certificateFile == "" && s.keyFile == "" {
+			l, err = srv.ListenTLS("tcp", s.ListenAddr)
+		} else {
+			// re-implement tailscale's Server.ListenTLS function with custom certificate for custom servers which do not offer a getCert function
+			cer, err := tls.LoadX509KeyPair(s.certificateFile, s.keyFile)
+			if err != nil {
+				return nil, fmt.Errorf("listener for %v: failed loading certificates: %w", srv, err)
+			}
+
+			ln, err := srv.Listen("tcp", s.ListenAddr)
+			if err != nil {
+				return nil, fmt.Errorf("listener for %v: %w", srv, err)
+			}
+
+			l = tls.NewListener(ln, &tls.Config{
+				Certificates: []tls.Certificate{cer},
+				MinVersion:   tls.VersionTLS12,
+			})
+		}
```

Comment:
Hmm, I think this might read nicer as another `case` instead of an `if` nested in this case. How's this, as a case branch right above `case !s.ServePlaintext`?

```go
	case s.certificateFile == "" && s.keyFile == "":
		// re-implement tailscale's Server.ListenTLS function with custom certificate for custom servers which do not offer a getCert function
		cer, err := tls.LoadX509KeyPair(s.certificateFile, s.keyFile)
		if err != nil {
			return nil, fmt.Errorf("listener for %v: failed loading certificates: %w", srv, err)
		}

		ln, err := srv.Listen("tcp", s.ListenAddr)
		if err != nil {
			return nil, fmt.Errorf("listener for %v: %w", srv, err)
		}

		l = tls.NewListener(ln, &tls.Config{
			Certificates: []tls.Certificate{cer},
			MinVersion:   tls.VersionTLS12,
		})
	case !s.ServePlaintext:
		l, err = srv.ListenTLS("tcp", s.ListenAddr)
```